### PR TITLE
add aria-label for inputs

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR adds the `aria-label` to the inputs in the DataFilter input 

HTML best practice for inputs is to provide labels, a fundamental principle outlined in the WCAG guidelines. It is possible to make an input accessible without the visible label. We can do this by using an Aria-label. The goal is to make the input clear to all users.

#### Where should the reviewer start?
datafilter.js
#### What testing has been done on this PR?
updated snapshots
#### How should this be manually tested?
use voice over on mac for how the inputs are read out for keyboard users. 
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
When DataFilter is used alone depending on the data it will use a `checkboxgroup` or a `range` these however do not have `labels` because in some cases they may be used in a `toolbar` for design purposes the `label` is left off. In order to still have these inputs accessible they need to have an `aria-label` associated with the input. 
#### What are the relevant issues? 
Data&Friends
Accessibility -- When DataFilter isn't wrapped in DataFilters, there's no label. Should it always be wrapped in DataFilters?
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no

### CODE REVIEW

File: src/js/components/DataFilter/DataFilter.js [hunk 1]
```diff
@@ -123,6 +123,7 @@ export const DataFilter = ({
     if (range) {
       content = (
         <RangeSelector
+          aria-label={property}
```
Comment (MikeKingdom) [inline src/js/components/DataFilter/DataFilter.js:126]: Shouldn't this do what the FormField would do for label and try to use a label defined in the properties? In many real world cases that property name isn't all that friendly/usable.
```jsx
label={properties?.[property]?.label || property}
```

**Review by MikeKingdom — COMMENTED** [review]
Should this only be in the `noForm` case? I would think the FormField `htmlFor` would take care of the connection to the label for screen readers.

File: src/js/components/DataFilter/DataFilter.js (hunk 1 above)
Comment (britt6612) [reply src/js/components/DataFilter/DataFilter.js:126]: good point yes the `property` isnt always the best

File: src/js/components/DataFilter/DataFilter.js [hunk 2]
```diff
@@ -118,11 +118,17 @@ export const DataFilter = ({
 
   const id = `${dataId}-${property}`;
 
+  // only add aria-label for no form examples
+  const ariaLabel = noForm
```
Comment (britt6612) [inline src/js/components/DataFilter/DataFilter.js:122]: open to naming suggestion if we wanted to change here

**Review by MikeKingdom — COMMENTED** [review]
Looking pretty good. Does the SelectMultiple need it too?

File: src/js/components/DataFilter/DataFilter.js [hunk 3]
```diff
@@ -118,11 +118,17 @@ export const DataFilter = ({
...
+    ? `${properties?.[property]?.label || property}`
+    : null;
```
[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/components/DataFilter/DataFilter.js:124]: Ill change

**Comment by britt6612** [comment]
> Looking pretty good. Does the SelectMultiple need it too?

SelectMultiple under the hood already has an `aria-label` with "open drop" so when I did the voice over I thought that was fine since that's our default behavior.

**Review by MikeKingdom — APPROVED** [review]

File: src/js/components/DataFilter/DataFilter.js [hunk 4]
```diff
@@ -144,10 +150,22 @@ export const DataFilter = ({
...
+          <CheckBoxGroup
+            aria-label={ariaLabel}
+            id={id}
+            name={property}
+            options={options}
+          />
+        );
       } else {
         content = (
           <SelectMultiple
```
Comment (taysea) [inline src/js/components/DataFilter/DataFilter.js:171]: `ariaLabel` seems to be missing from here?

File: src/js/components/DataFilter/DataFilter.js (hunk 2 above)
Comment (taysea) [reply src/js/components/DataFilter/DataFilter.js:122]: This variable name is fine to me

File: src/js/components/DataFilter/DataFilter.js (hunk 4 above)
Comment (britt6612) [reply src/js/components/DataFilter/DataFilter.js:171]: That's fine I can add it here.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**